src/main/flight/pid.c

```diff
@@ -740,6 +741,48 @@ float pidGetAirmodeThrottleOffset(void)
 }
 #endif
 
+// when new Rx data is received, update the maxDeflectionAbs value, so we don't do this every PID loop
+static float maxDeflectionAbs = 0.0f;
+float sendMaxDeflectionAbs(float maxRcDeflectionAbs)
+{
+    maxDeflectionAbs = fmaxf(maxRcDeflectionAbs, mixerGetRcThrottle());
+    return maxDeflectionAbs;
+}
+
+static void disarmOnImpact(const int axis)
+{
+    // if both sticks are inside 20% of the stick threshold, enable auto-disarm on impact
+    // will not disarm on gentle landings
+    // value should be highe enough to avoid unwanted disarms in the air on throttle chops
+    // note that unlike GPS code, this just compares one axis at a time, to reduce CPU load
+    // for the Z axis, 1G gets added while 'flat', but that won't matter much given we are looking around 8G mostly
+    float accMagnitude = fabsf(acc.accADC[axis]) * acc.dev.acc_1G_rec;
+    // *** annoying to need to normalise accADC here, should normalise acc.accADC once, in accelerometer.c, not everywhere we use it
+    if (isAirmodeActivated() && maxDeflectionAbs < pidRuntime.ezLandingThreshold * 0.5f) {
+        if (accMagnitude > pidRuntime.ezLandingDisarmThreshold) {
```

Comment:
This can trigger on high accelerations (which are intended and not meant to be crashes). Acceleration is not a good measure for detecting collisions. Jerk - the derivative of acceleration - is better suited for that.

Reply:
Earlier versions used the method in GPS Rescue, with root sum of squares, and that worked well.  I didn't want to use a method that we knew worked reliably enough in GPS Rescue.  This code would definitely make an easy place to test whether derivative of acceleration would be useful.  Some raw accelerometer data can be very noisy, and that may affect the derivative and cause false triggering.

Reply:
Went back to sort(sum of squares) - 1, and check for. disarm only once per loop.